src/routes/viewer/CourseCombobox.tsx

```diff
@@ -17,10 +17,12 @@ import CustomMap from "@/utils/CustomMap";
 import { CourseInfo } from "@/utils/data/store";
 import { useState } from "react";
 
+type onSelectType = (value: string) => void;
+
 type Props = {
   courses: CustomMap<string, CourseInfo>;
   value: string;
-  onSelect: (value: string) => void;
+  onSelect: onSelectType;
```

Comment:
Questa cosa (che hai fatto un po' dappertutto) nel frontend (specialmente in react) è molto sconsigliata, perché
1) rallenta molto il processo di typing
2) induce a errori di reference e di update
3) moltiplica il numero di interfacce e rende l'applicazione più lenta